src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -141,10 +142,11 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
     def privates[T <: Tree](typer: Typer, owner: Symbol, tree: T): T =
       if (owner.isJavaDefined) tree else check(typer, owner, EmptyScope, WildcardType, tree)
 
+    @tailrec
     private def check[T <: Tree](typer: Typer, owner: Symbol, scope: Scope, pt: Type, tree: T): T = {
       this.owner = owner
       this.scope = scope
-      hiddenSymbols = List()
+      hiddenSymbols = Nil
```

Comment:
OK but scalac does this for you. It seems to me there are previous PRs about this, but I don't remember which is preferred.

Reply:
I find it more readable but maybe others disagree

Reply:
I don't disagree.